mypy/newsemanal/semanal_pass1.py

```diff
@@ -44,17 +47,35 @@ def visit_file(self, file: MypyFile, fnam: str, mod_id: str, options: Options) -
         self.is_global_scope = True
 
         for i, defn in enumerate(file.defs):
-            if isinstance(defn, (ClassDef, FuncDef)):
-                self.is_global_scope = False
             defn.accept(self)
-            self.is_global_scope = True
             if isinstance(defn, AssertStmt) and assert_will_always_fail(defn, options):
                 # We've encountered an assert that's always false,
                 # e.g. assert sys.platform == 'lol'.  Truncate the
                 # list of statements.  This mutates file.defs too.
                 del file.defs[i + 1:]
                 break
 
+    def visit_func_def(self, node: FuncDef) -> None:
+        old_global_scope = self.is_global_scope
+        self.is_global_scope = False
+        super().visit_func_def(node)
+        self.is_global_scope = old_global_scope
+        file_node = self.cur_mod_node
+        if (self.is_global_scope
+                and file_node.is_stub
+                and node.name() == '__getattr__'
+                and file_node.is_package_init_file()):
+            # __init__.pyi with __getattr__ means that any submodules are assumed
+            # to exist, even if there is no stub. Note that we can't verify that the
+            # return type is compatible, since we haven't bound types yet.
```

Comment:
This is OK, but I think it is something worth documenting.

Reply:
This could be mentioned in mypy release notes when we announce the new semantic analyzer.